common/version-control/src/main/java/org/thingsboard/server/service/sync/vc/DefaultGitRepositoryService.java

```diff
@@ -195,8 +195,17 @@ public List<BranchInfo> listBranches(TenantId tenantId) {
     }
 
     private GitRepository checkRepository(TenantId tenantId) {
-        return Optional.ofNullable(repositories.get(tenantId))
+        GitRepository gitRepository = Optional.ofNullable(repositories.get(tenantId))
                 .orElseThrow(() -> new IllegalStateException("Repository is not initialized"));
+
+        if (!Files.exists(Path.of(gitRepository.getDirectory()))) {
+            try {
+                return cloneRepository(tenantId, gitRepository.getSettings());
+            } catch (Exception e) {
+                throw new IllegalStateException("Repository is not initialized");
```

Comment:
-> `new IllegalStateException("Could not initialize the repository: " + e.getMessage(), e)`